pyomo/core/base/var.py

```diff
@@ -371,6 +388,16 @@ def value(self):
     @value.setter
     def value(self, val):
         """Set the value for this variable."""
+        if type(val) in native_numeric_types:
```

Comment:
This is not a problem with this PR, but `value` and `set_value` should do the same thing.